handlers/developers_handler.js

```diff
@@ -47,27 +14,110 @@ var DeveloperHandler = {
+    })
+  },
+  /**
+   * Sets the response object to be in the format of an error.
+   * @param {object} res - The response object
+   * @param {string} err - The specific error
+   * @param {Number} statusCode - The http status code
+   */
+  setErrorResponse: function (res, err, statusCode) {
+    res.status(statusCode)
+    res.format({
+      json: function () {
+        res.json({error: err})
+      }
+    })
+  },
+  /**
+   * Gets a users information based on a username parameter.
+   * @param {String} username - A username in the database
+   * @returns {Promise} If successful, the username and corresponding token is returned
+   */
+  getValue: function (username) {
+    return new Promise((resolve, reject) => {
+      this.client.get(username, function (err, token) {
+        if (err) {
+          reject(err)
+        }
+        resolve({
+          username,
+          token: token
+        })
+      })
+    })
+  },
+  /**
    * @param {Object} req - The request object, containing the username and the token in the query
    * @param {Object} res - The response object, which will be send back to the front end after the request has been returned.
    *                       It contains the status code and the json with either the username and token or the cooresponding error.
    * @returns {Object} err - This function will always resolve with the err object, since we are required to either resolve or reject the promise.
    */
   get: function (req, res) {
     return new Promise((resolve, reject) => {
-      this.client.get(req.query.username, function (err, redisResp) {
-        if (err) {
-          setErrorResponse(res, err, 404)
-        } else if (redisResp === null) {
-          setErrorResponse(res, 'The username provided does not match any username. Please make sure that you are signed up as an authorized ciena developer on www.cienadevelopers.com', 500)
-        } else if (req.query.token === '') {
-          setStandardResponse(res, req.query.username, redisResp)
-        } else if (req.query.token === redisResp) {
-          setStandardResponse(res, req.query.username, redisResp)
-        } else {
-          setErrorResponse(res, 'The token submitted does not match the token returned.', 500)
-        }
-        resolve(err)
-      })
+      let pset = []
+      if (req.query.queryAll) {
+        this.client.keys('*', function (err, keys) {
+          if (err) {
+            DeveloperHandler.setErrorResponse(res, err, 404)
+          } else {
+            for (let key in keys) {
+              pset.push(DeveloperHandler.getValue(keys[key]))
+            }
+            Promise.all(pset).then((values) => {
+              DeveloperHandler.setStandardKeysResponse(res, values)
+            }, (reason) => {
+              DeveloperHandler.setErrorResponse(res, reason, 404)
+            })
+          }
+          resolve(keys)
+        })
+      } else {
+        this.client.get(req.query.username, function (err, redisResp) {
```

Comment:
I find this `if... else` structure a bit rambling for a single function. I believe this code should be factored out into its own function for clarity sake.